pkg/services/cosmosdb/cosmosdb.go

```diff
@@ -23,8 +23,7 @@ type sqlAccountManager struct {
 }
 
 type mongoAccountManager struct {
-	armDeployer            arm.Deployer
-	databaseAccountsClient cosmosSDK.DatabaseAccountsClient
+	cosmosAccountManager
```

Comment:
I was momentarily concerned that the polymorphic behavior of certain functions having a `mongoAccountManager ` receiver wouldn't properly "override" the functions of the same name having the composed `cosmosAccountManager` as a receiver, but I read up and also crafter an example in the playground and it all checks out. This is a nice improvement.